tuw_nlp/common/eval.py

```diff
@@ -60,25 +60,35 @@ def count_p_r_f(cat_stats):
     return cat_stats
 
 
-def get_cat_stats(preds, golds):
+def get_cat_stats(preds, golds, labels_to_keep=None, add_missing_labels=False):
+    assert not (add_missing_labels and labels_to_keep is None)
     stats = defaultdict(Counter)
     for pred, gold in zip(preds, golds):
         p = set(pred)
         g = set(gold)
-        for label in p & g:
-            stats[label]["TP"] += 1
-        for label in g - p:
-            stats[label]["FN"] += 1
-        for label in p - g:
-            stats[label]["FP"] += 1
-
+        _add_to_stat(stats, "TP", g & p, labels_to_keep)
+        _add_to_stat(stats, "FN", g - p, labels_to_keep)
+        _add_to_stat(stats, "FP", p - g, labels_to_keep)
+    if add_missing_labels:
+        for label in labels_to_keep:
+            if label not in stats:
+                stats[label] = {
```

Comment:
the other values of `stats` are `Counter` objects, so please create it as such, e.g. `stats[label] = Counter({"TP": 0, ...`, to keep it consistent

Reply:
How about the solution in [5c34920](https://github.com/recski/tuw-nlp/pull/76/commits/5c34920b3fa9cbe653d925826b07b7f64bde2ba7)?

Reply:
even better, thank you!